docs/flow/modin/experimental/engines/omnisci_on_ray/omnisci_engine.rst

```diff
@@ -0,0 +1,135 @@
+:orphan:
+
+OmniSciDB Execution Engine (experimental)
+=========================================
+
+OmniSciDB is an open-source SQL-based relational database designed for the
+massive parallelism of modern CPU and GPU hardware. Its execution engine
+is built on LLVM JIT compiler.
+
+OmniSciDB can be embedded into an application as a dynamic library that
+provides both C++ and Python APIs. A specialized in-memory storage layer
+provides an efficient way to import data in Arrow table format.
+
+Relational engine limitations
+-----------------------------
+
+Using a relational database engine implies a set of restrictions on
+operations we can execute on a dataframe.
+
+1. We cannot handle frames that use data types not supported by OmniSciDB.
+Currently, we allow only integer, float, string, and categorical data types.
+
+2. Column data should be homogeneous.
+
+3. Can only support operations that map to relational algebra. This means
+most operations are supported over a single axis (axis=0) only. Non-relational
+operations like transposition and pivot are not supported.
+
+When the unsupported data type is detected or unsupported operations is requested
+we use the original pandas framework.
+
+Lazy execution
+--------------
+
+OmniSciDB has a powerful query optimizer and an execution engine that
+combines multiple operations into a single execution module. E.g. join,
+filter and aggregation can be executed in a single data scan.
+
+To utilize this feature and reduce data transformation and transfer
+overheads, we need to implement lazy operations on a dataframe. The
+dataframe with lazy computation is implemented in
+:py:class:`~modin.experimental.engines.omnisci_on_ray.frame.data.OmnisciOnRayFrame`
+class.
+
+Lazy operations on a frame build a tree which is later translated into
+a query executed by OmniSci. We use two types of trees. The first one
+describes operations on frames that map to relational operations like
+projection, union, etc. Nodes in this tree are derived from ``DFAlgNode``
```

Comment:
This comment ([link](https://github.com/modin-project/modin/pull/2984#discussion_r655254987)) is applicable to all similar things in rst documentation (DFAlgNode, TransformNode, etc.).

Reply:
OK. Will fix that. Do external classes like `pandas.DataFrame` and `pyarrow.Table` need some additional formatting?

Reply:
If you want to use this style pandas.DataFrame/pyarrow.Table, double backticks should be enough.
If you want to use this style pandas DataFrame/PyArrow Table, we can use it with no any backtick.